### DISCUSSION TITLE
Unlocking TurboWarp’s Potential: Feature Enhancements

### DISCUSSION THREAD

**Discussion by Eaielectronic** [TurboWarp/extensions#1700] [Q&A]
Hello !

I would like to share an idea that I believe could greatly enhance the user experience on TurboWarp. Imagine a system where, when you load an extension from a local file, TurboWarp automatically updates that extension if the file is modified. This would allow for easy addition of blocks without the need to delete and reload the extension.

Additionally, if the file no longer exists, TurboWarp should not change the extension when loading the project, ensuring continuity in development. I would also like to propose the integration of a mini Visual Studio Code-like editor for smoother management of extensions.

It would also be very helpful to add a "without sandbox" option for URLs, allowing users to use trusted GitHub repositories without restrictions. Lastly, if an extension with the same ID is added via text, the previous version could be overwritten while keeping the block positioning intact, thus facilitating project updates.

I am convinced that TurboWarp has the potential to become an extremely popular coding tool due to its simplicity and the speed of game creation. By making the desktop software more open to handling JavaScript, HTML, and extensions, somewhat like a mini Visual Studio Code, we could really attract an even larger audience.

Thank you for taking the time to read my suggestions. I look forward to your feedback and am available to discuss these ideas further!

Respectfully.

**Comment by Thebloxers998** [comment]
Sup bro
